### DISCUSSION TITLE
Page list empty when creating Facebook channel

### DISCUSSION THREAD

**Discussion by miamicityman** [chatwoot/chatwoot#7491] [Self hosted] [answered]
I have reinstalled from scratch several times and have read the following numerous times.

https://www.chatwoot.com/docs/self-hosted/configuration/features/integrations/facebook-channel-setup

I can't get messenger to work on Chatwoot Self-hosted.

![Screenshot_1](https://github.com/chatwoot/chatwoot/assets/57761794/b05ae4d3-9312-43a8-b9cc-1084a2b3306a)

I see that I'm not the only person who's had this issue, but the prior post is left unanswered.

It logs on correctly but when I return to Chatwoot, the pull-down is blank.

I'm using **Facebook Login for Business** as FB developer app but all the parameters are the same per instructions.

Any ideas anyone?

**Comment by miamicityman** [comment]
Hello? Anyone?

**Reply by jtayrone** [reply]
For the messenger to work you need to have an app approved in the meta with the permissions listed in the [documentation](https://www.chatwoot.com/docs/self-hosted/configuration/features/integrations/facebook-channel-setup). If it's just a test, just follow the procedure in the [documentation](https://www.chatwoot.com/docs/self-hosted/configuration/features/integrations/facebook-channel-setup).

**Reply by uhlhosting** [reply]
@jtayrone i have myself followed the docs for testing and i only got 3 pages from 7 shown. I guess might be that they are business apps too. And maybe the testing as its now wont work with those apps without having live the fb app.

**Comment by uhlhosting** [comment]
I got the same issue and reported to the team on discord, we need to wait, I think everyone is busy to release v3

@sojan-official @vishnu-narayanan Any idea on what could cause this?

More details I added on: https://discord.com/channels/647412545203994635/1132618395595067402

**Reply by LiLiJal** [reply]
Heey Viorel, 
I can see that on GIthub there are many people encountering the same issue with FB channel not showing the pages or showing  some of the pages, but no answer so far( are you guys working on that or it cannot be fixed? As i thing the issue is with old pages

**Comment by pedrofnts** [comment]
bump @sojan-official @vishnu-narayanan

**Comment by kcalliauw** [comment]
I'm having the same issue

**Comment by mredodos** [comment]
first time after i have add facebook all works good, after i have add instagram integration trigger that problem in this post how can we solve?

EDIT: i think is empty maybe because exist as inbox in the database but not appear in the frontend

**Comment by pierrecorsini** [comment]
Same issue here v3.3.1 Build dcd18072b49cb8176c8686f70584fc75064d0c71

**Reply by pranavrajs** [reply]
@miamicityman You need to get the business_management permission for it to work. We will update the docs soon.

**Reply by pierrecorsini** [reply]
There should be some other parameters linked to this issue. Adding `business_management ` I can 1/3 of my FB's account Pages, others are missing. (P.S. 4 fields as the messaging_postbacks was not added by mistake).
![image](https://github.com/chatwoot/chatwoot/assets/50719398/54cc9859-5ce0-4e6d-a10e-e6285f3cb75d)

**Reply by pierrecorsini** [reply]
Also I do confirm that the following shows all my page:

- Open https://developers.facebook.com/tools/explorer
- Require token for business management, and page show list permissions.
- Call me/accounts
- => All the pages are listed on the response.

**Reply by pierrecorsini** [reply]
Well this seems that business_management is indeed in the code and on my server at `app/javascript/dashboard/routes/dashboard/settings/inbox/channels/Facebook.vue` **but not in my front end** (not sure if it's related to cache or failed cwctl update). Doing some JS hijacking I am able to add business_management in the FB login scope and this then works...

**Reply by vishnu-narayanan** [reply]
@pierrecorsini That is interesting. Could you check if the cwctl upgrade was successful?  Also could you share the commit id your instance is at? The `git sha` should be available at https://<your-instance-url>/super_admin/instance_status

**Reply by pierrecorsini** [reply]
```
Metric	Value
Chatwoot version	3.3.1
Git SHA	dcd18072b49cb8176c8686f70584fc75064d0c71
Postgres alive	true
Redis alive	true
Redis version	7.2.2
Redis number of connected clients	20
Redis 'maxclients' setting	10000
Redis memory used	2.92M
Redis memory peak	3.08M
Redis total memory available	860.96M
Redis 'maxmemory' setting	0
Redis 'maxmemory_policy' setting	noeviction
Chatwoot edition	Enterprise
Database Migrations	completed
```

```
~/chatwoot$ git show
commit dcd18072b49cb8176c8686f70584fc75064d0c71 (HEAD -> master, tag: v3.3.1, origin/master)
Merge: bda2d1d9a 4244945e7
Author: Sojan <sojan@pepalo.com>
Date:   Sat Nov 18 10:24:25 2023 -0800

    Merge branch 'release/3.3.1'
```

There could be an issue with the update; I logged this about it : https://github.com/chatwoot/chatwoot/issues/8485

P.S. I also tried to git reset hard (as I had some diff due to asset compilation , to be sure git was clean before update) but this changes nothing.

**Comment by pedrofnts** [comment]
So, was this solved in Chatwoot latest version?

**Reply by pierrecorsini** [reply]
No report available about it working or not working. Have you tried ?

**Reply by pedrofnts** [reply]
Nope, didn't updated yet.

**Comment by NomiIqbal** [comment]
Hi @pranavrajs @tejaswinichile @vishnu-narayanan 

guyz need Use Case and screencast to approve "business_management" permission. 
FB rejected every time request with invalid use case.

Also anyone have approved "business_management" permission in chatwoot community? please share the details it will be helpful.
Thank You!

**Reply by mredodos** [reply]
I can't get it validated in any way I want pictures and videos but you can't even tell which ones

**Comment by ben-pr-p** [comment]
I had this issue on Chatwoot 2.2.x and then when I upgraded to 3.11.0 and recreated the inbox it was fixed

**Comment by mredodos** [comment, marked as answer]
yes in the last version it seems fixed

**Reply by Achrafchrouda** [reply]
can someone tell me what is commit that is fix the bug ? so i can added in my version ! 
i dont wanna upgrade my version

**Comment by mredodos** [comment]
With last 2 version i have updated withiut problems

Il Mar 3 Set 2024, 16:52 Achrafchrouda ***@***.***> ha
scritto:
